core/store/src/db/rocksdb.rs

```diff
@@ -167,52 +167,51 @@ impl RocksDB {
     ) -> RocksDBIterator<'a> {
         let cf_handle = self.cf_handle(col);
         let mut read_options = rocksdb_read_options();
-        let mode = if let Some(prefix) = prefix {
-            // prefix_same_as_start doesn’t do anything for us.  It takes effect
-            // only if prefix extractor is configured for the column family
-            // which is something we’re not doing.  Setting this option is
-            // therefore pointless.
+        if let Some(prefix) = prefix {
+            read_options.set_iterate_range(::rocksdb::PrefixRange(prefix));
+            // Note: prefix_same_as_start doesn’t do anything for us.  It takes
+            // effect only if prefix extractor is configured for the column
+            // family which is something we’re not doing.  Setting this option
+            // is therefore pointless.
             //     read_options.set_prefix_same_as_start(true);
-
-            // We’re running the iterator in From mode so there’s no need to set
-            // the lower bound.
-            //    read_options.set_iterate_lower_bound(key_prefix);
-
-            // Upper bound is exclusive so if we set it to the next prefix
-            // iterator will stop once keys no longer start with our desired
-            // prefix.
-            if let Some(upper) = next_prefix(prefix) {
-                read_options.set_iterate_upper_bound(upper);
-            }
-
-            IteratorMode::From(prefix, Direction::Forward)
-        } else {
-            IteratorMode::Start
-        };
-        let iter = self.db.iterator_cf_opt(cf_handle, read_options, mode);
-        RocksDBIterator(Some(iter))
+        }
+        let iter = self.db.iterator_cf_opt(cf_handle, read_options, IteratorMode::Start);
+        RocksDBIterator(iter)
     }
 }
 
-struct RocksDBIterator<'a>(Option<rocksdb::DBIteratorWithThreadMode<'a, DB>>);
+struct RocksDBIterator<'a>(rocksdb::DBIteratorWithThreadMode<'a, DB>);
 
 impl<'a> Iterator for RocksDBIterator<'a> {
     type Item = io::Result<(Box<[u8]>, Box<[u8]>)>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let iter = self.0.as_mut()?;
-        if let Some(item) = iter.next() {
-            Some(Ok(item))
-        } else {
-            let status = iter.status();
-            self.0 = None;
-            status.err().map(into_other).map(Result::Err)
-        }
+        Some(self.0.next()?.map_err(into_other))
     }
 }
 
 impl<'a> std::iter::FusedIterator for RocksDBIterator<'a> {}
 
+impl RocksDB {
+    fn get_cf_key_range(
+        &self,
+        cf_handle: &ColumnFamily,
+    ) -> Result<Option<std::ops::Range<Box<[u8]>>>, ::rocksdb::Error> {
```

Comment:
```suggestion
    ) -> Result<Option<std::ops::RangeInclusive<Box<[u8]>>>, ::rocksdb::Error> {
```

I *think* this is inclusive?